### DISCUSSION TITLE
Need help with 8BIM Profile , how to create new one from scratch ?

### DISCUSSION THREAD

**Discussion by misko2k** [dlemstra/Magick.NET#1341] [Help]
Hi ,

I have picture without clipping mask, and have tool what analize picture and depending on the result i must draw clipping mask.
so far so good, but how to get this clipping path into my picture ? 

Have read that 8BIM (EightBIM) must be created, but how ...

btw. i use C# with Magick.NET -Q8-AnyCP

any help are appreciated

**Comment by dlemstra** [comment]
There is no build-in tooling in Magick.NET or ImageMagick that can be use to create that in the 8BIM profile. You will need to do the inverse of the internal `ClipPathReader`. I cannot find the specifications that I used to create that class so I cannot really help you with that now.

**Reply by misko2k** [reply]
I have so far decoded the 8BIM Record, now i must read only coodinates from 26 byte resource blocks. Now i must find out how to convert HEX to float, according this:

`All points used in defining a path are stored in eight bytes as a pair of 32-bit components, vertical component first.
The two components are signed, fixed point numbers with 8 bits before the binary point and 24 bits after the binary point. Three guard bits are reserved in the points to eliminate most concerns over arithmetic overflow. Hence, the range for each component is 0xF0000000 to 0x0FFFFFFF representing a range of -16 to 16. The lower bound is included, but not the upper bound.

This limited range is used because the points are expressed relative to the image size. The vertical component is given with respect to the image height, and the horizontal component is given with respect to the image width. [ 0,0 ] represents the top-left corner of the image; [ 1,1 ] ([ 0x01000000,0x01000000 ]) represents the bottom-right.

In Windows, the byte order of the path point components are reversed; you should swap the bytes when accessing each 32-bit value`

Something there i am doing wrong, cause i get garbage out and not expected value.

Can you please look how you are encoded this coordinates ?
